code/datums/components/bump_attack.dm

```diff
@@ -2,34 +2,38 @@
 	var/active = TRUE
 	var/bump_action_path
 	var/datum/action/bump_attack_toggle/toggle_action
+	var/mob/living/bumper
 
 
 /datum/component/bump_attack/Initialize()
 	. = ..()
 	if(!isliving(parent))
 		return COMPONENT_INCOMPATIBLE
-	toggle_action = new()
-	var/toggle_path
-	if(ishuman(parent))
+	bumper = parent
+	if(ishuman(bumper))
 		bump_action_path = .proc/human_bump_action
-	else if(isxeno(parent))
+	else if(isxeno(bumper))
 		bump_action_path = .proc/xeno_bump_action
 	else
 		bump_action_path = .proc/living_bump_action
+	if(active)
+		RegisterSignal(bumper, COMSIG_MOVABLE_BUMP, bump_action_path)
+	var/toggle_path
 	toggle_path = .proc/living_activation_toggle
-	toggle_action.give_action(parent)
+	toggle_action = new()
+	toggle_action.give_action(bumper)
 	toggle_action.update_button_icon(active)
 	RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, toggle_path)
-	if(active)
-		RegisterSignal(parent, COMSIG_MOVABLE_BUMP, bump_action_path)
 
 /datum/component/bump_attack/Destroy(force, silent)
+	if(active)
+		UnregisterSignal(bumper, COMSIG_MOVABLE_BUMP)
 	QDEL_NULL(toggle_action)
+	bumper = null
```

Comment:
Don't take my word for it, but i think you also need to clean that ref when bumper is QDEL. Ask tivi about this, I don't think you are protecting this from hard del enough. It's possible that the component is automaticly destroyed on the parent destroy.

Reply:
removed var